src/transaction/transaction.rs

```diff
@@ -71,9 +79,38 @@ impl Transaction {
     /// ```
     pub async fn batch_get(
         &self,
-        _keys: impl IntoIterator<Item = impl Into<Key>>,
+        keys: impl IntoIterator<Item = impl Into<Key>>,
     ) -> Result<Vec<KvPair>> {
-        unimplemented!()
+        let mut result = Vec::new();
+        let mut keys_from_snapshot = Vec::new();
+        let mut result_indices_from_snapshot = Vec::new();
+
+        // Try to fill the result vector from mutations
+        for key in keys {
+            let key = key.into();
+            if let Some(value) = self.get_from_mutations(&key) {
+                result.push((key, value).into());
+            } else {
+                keys_from_snapshot.push(key);
+                result_indices_from_snapshot.push(result.len());
+                // Push a placeholder
+                result.push(KvPair::default());
```

Comment:
Finally we have to return a `Vec<KvPair>`. If we use `Option`, we need one more `Vec`.

Reply:
I think that we should be returning a pair of a Key and Option<Value>, and that that would be fine to use here too. There might be opportunity later to do some interesting optimisations here (if we end up using streams, or with whatever the type signatures end up looking like).

Anyway, I think it is better to use an `Option` for now since that is the 'right' thing to do, using a null value is premature optimisation, IMO. We can optimise later.